### PR TITLE
std::monostate should act like nullptr

### PR DESCRIPTION
**Pull request by epsilon-phase**
Okay, so, not what you asked yet, but, this is a better start for this pull request.

So, let's see

- [x] Documentation in readme
- [x] Testcase that expresses its use
- [x] **mystery meat**???
- [x] Dev branch MAGICK

### CODE REVIEW

**Comment by aminroosta** [comment]
@epsilon-phase Have a look at [has_sqlite_type<>](https://github.com/SqliteModernCpp/sqlite_modern_cpp/blob/dev/hdr/sqlite_modern_cpp/type_wrapper.h#L161) and [is_sqlite_value<>](https://github.com/SqliteModernCpp/sqlite_modern_cpp/blob/dev/hdr/sqlite_modern_cpp/type_wrapper.h#L60)

**Comment by epsilon-phase** [comment]
This seems to be working fine for me at this point, are there any remaining concerns that you'd like me to address?

File: hdr/sqlite_modern_cpp/type_wrapper.h [hunk 1]
```diff
@@ -166,6 +166,26 @@ namespace sqlite {
 	inline void store_result_in_db(sqlite3_context* db, std::nullptr_t) {
 		sqlite3_result_null(db);
 	}
+#ifdef MODERN_SQLITE_STD_VARIANT_SUPPORT
+	template<>
+	struct has_sqlite_type<std::monostate, SQLITE_NULL, void> : std::true_type {};
+  
+	inline std::monostate get_col_from_db(sqlite3_stmt* stmt, int inx, result_type<std::monostate>) {
+    if(sqlite3_column_type(stmt,inx) != SQLITE_NULL) throw 1;
```
Comment (aminroosta) [inline hdr/sqlite_modern_cpp/type_wrapper.h]: @zauguin This doesn't look right to me, should we just return `std::monostate` all the time in `get_col_from_db(sqlite3_stmt*, in, result_type<std::monostate>` ?

Comment (epsilon-phase) [reply hdr/sqlite_modern_cpp/type_wrapper.h]: I wasn't sure about whether or not this was necessary so this is more from making sure that I could figure out what'd gone wrong.

I've removed it and found no problems, so I'm inclined to commit that version instead.

Comment (aminroosta) [reply hdr/sqlite_modern_cpp/type_wrapper.h]: @zauguin Is it good to merge? you know more than me about this part.

**Comment by zauguin** [comment]
@epsilon-phase @aminroosta Looks good to me, but I would prefer if the 3 unrelated whitespace changes would be omitted.

**Comment by epsilon-phase** [comment]
So, remove the whitespace changes?
Alright

**Pull request merged**